src/op.rs

```diff
@@ -1198,34 +1266,210 @@ impl<'context, 'input, Endian, Context> Evaluation<'context, 'input, Endian, Con
             }
 
             Operation::EntryValue { expression } => {
-                let value = try!(self.callbacks.evaluate_entry_value(expression.into()));
-                self.push(value);
+                return Ok(OperationEvaluationResult::AwaitingEntryValue {
+                    expression: expression.into(),
+                });
             }
 
             Operation::Piece { .. } => {
                 piece_end = true;
             }
         }
 
-        Ok((piece_end, terminated, current_location))
+        Ok(OperationEvaluationResult::Complete {
+            terminated: terminated,
+            piece_end: piece_end,
+            current_location: current_location,
+        })
     }
 
-    /// Evaluate a DWARF expression.  If successful, the result will
-    /// hold a vector of pieces, each describing a part of the final
-    /// object.
-    pub fn evaluate(&mut self) -> Result<Vec<Piece<'input>>, Context::ContextError>
-        where Endian: Endianity
+    /// Get the evaluation result.  This `Evaluation` must have previously been
+    /// driven to completion by calling the evaluate and resume_with methods
+    /// until `EvaluationResult::Complete` was returned.
```

Comment:
This should have a section of the docs like the standard library does about panics, for example like in here: https://doc.rust-lang.org/nightly/std/string/struct.String.html#method.insert_str

So it would be something like this:

```markdown
# Panics

If this `Evaluation` has not been driven to completion, calling `result` will panic.
```